docs/odh/fetch-upstream-release.md

```diff
@@ -0,0 +1,430 @@
+kserve  git@github.com:kserve/kserve.git (fetch)
+kserve  git@github.com:kserve/kserve.git (push)
+odh     git@github.com:opendatahub-io/kserve.git (fetch)
+odh     git@github.com:opendatahub-io/kserve.git (push)
+origin  git@github.com:${GH_USER}/kserve.git (fetch)
+origin  git@github.com:${GH_USER}/kserve.git (push)
+```
+
+> [!NOTE]  
+> Although in this setup the remotes are configured with SSH, you can also use
+> HTTPS if that is what you prefer.
+
+## Performing the branch cut
+
+Fetch the tags of the community repository:
+
+```sh
+$ git fetch kserve --tags
+```
+
+Double check that the tag in your working copy is up to date with the tag in the
+upstream repository
+
+```sh
+VERSION_TO_PULL="v0.15.0" # Replace with the kserve/kserve tag you want to push to ODH
+LOCAL_SHA=$(git log -1 $VERSION_TO_PULL --pretty=format:"%H")
+REMOTE_SHA=$(curl -s -H "Accept: application/vnd.github+json" https://api.github.com/repos/kserve/kserve/commits/$VERSION_TO_PULL | jq -r '.sha')
+[[ "$LOCAL_SHA" == "$REMOTE_SHA" ]] && echo "OK! Commits match." || echo "Commits does not match."
+```
+
+From the previous script, ensure you get the "OK! Commits match." message.
+Otherwise, stop and review your setup.
+
+Fetch `master` branch from
+[opendatahub-io/kserve](https://github.com/opendatahub-io/kserve/) repository:
+
+```sh
+$ git fetch odh master
+```
+
+Validate that you can proceed with the branch cut:
+
+```sh
+$ (git branch -a --contains tags/${VERSION_TO_PULL} | grep 'remotes/odh/master') && echo "Tag is available to fetch" || echo "Code sync is required"
+```
+
+If the output of the previous command is `Code sync is required`, a code sync is
+must be done before the branch cut can be done. If the expected output `Tag is
+available to fetch` is printed, the next step is to resolve what commit can be
+used for the cut. 
+
+Let's assume the following simplified graph of the `master` branches of ODH and
+upstream may look like in the following graph:
+
+    odh/master                *---*---*---A---*---*---*---B---*---*---*---*---C  (most recent commit)
+                             /           /               /           /
+    kserve/master   *---*---*---*---*---α---*---β---*---γ---*---*---*---*---δ  (most recent commit)
+                                                |
+                                            tag/release
+
+The letters and the `*` character would represent commits on each fork's
+`master` branches. The slash `/` characters would represent merges (i.e. code
+syncs) to ODH fork that integrate code from upstream. Notice that the upstream
+release `β` happened between the code sync `A` and `B`. It must be noted that
+despite commit `β` can be reached from any commit after `B` (inclusive) on
+odh/master, the ODH commits/customizations cannot be reached from the commit `β`
+(i.e. the upstream release tag). This is because merges are one way: upstream ->
+ODH. Should we generate a branch from commit `β`, we would not see any ODH
```

Comment:
`Should we generate` it is not clear what it means, is it a question?

Reply:
I see, for me, when we start something with "Should we" generally is a question, but I am not native, so please disregard my comment :D